### PR TITLE
issues-1248: ArtifactManagementServiceImplTest.testConcurrentReadWrite  

### LINKED ISSUE #1248: Investigate why the ArtifactManagementServiceImplTest is failing
# Task Description

We've had to disable the `ArtifactManagementServiceImplTest`, as a follow-up to #1246. We need to re-enable this test and resolve it's issues with high priority. This test needs to be re-worked to use the new test annotations, as explained in #1165. Once the test is re-enabled, it should start failing in our Travis build, as Travis is building the code under MacOS, where it seems to almost always be reproducible.

# Acceptance Test

* [ ] Building the code with `mvn clean install -Dintegration.tests` still works.
* [ ] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [ ] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [ ] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Task Relationships

This task:
* Is a sub-task of: #1165 
* Is a follow-up of: #1246
* Relates to: #1101 , #1243 , #1165

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

### PR DESCRIPTION
**Pull request by sbespalov**
Fixes #1248

- resource lock fixed within `RepositoryInputStream`/`RepositoryInputStream` with `LazyInputStream`/`LazyOutputStream`
- concurrent tests logic improved

### CODE REVIEW

File: strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/io/LazyInputStream.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
+package org.carlspring.strongbox.io;
+
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * @author sbespalov
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/io/LazyInputStream.java]: Would you mind adding some comments/documentation of when these two "lazy" streams should be used and what's special about them? Thanks!

Comment (sbespalov) [reply strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/io/LazyInputStream.java]: done

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/StorageFileSystemProvider.java [hunk 2]
```diff
@@ -307,19 +318,23 @@ public void undelete(RepositoryPath path)
     public RepositoryPath moveFromTemporaryDirectory(TempRepositoryPath tempPath)
         throws IOException
     {
+        logger.debug(String.format("Mooving [%s]", tempPath.getTarget()));
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/StorageFileSystemProvider.java]: Cosmetic: There's a typo in "Moving".

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/resources/junit-platform.properties [hunk 3]
```diff
@@ -1,3 +1 @@
-junit.jupiter.execution.parallel.enabled=true
-junit.jupiter.execution.parallel.config.strategy=fixed
-junit.jupiter.execution.parallel.config.fixed.parallelism=2
+junit.jupiter.execution.parallel.enabled=true
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/resources/junit-platform.properties]: Would you mind clarifying why you've removed these properties? Thanks! :)

**Comment by carlspring** [comment]
LGTM, with minor comments.

Would you mind rebasing? Thanks! :)

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/resources/junit-platform.properties (hunk 3 above)
Comment (sbespalov) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/resources/junit-platform.properties]:
```
junit.jupiter.execution.parallel.config.strategy=fixed
junit.jupiter.execution.parallel.config.fixed.parallelism=2
```

this settings affects on behavior of `BaseStream.parallel()` from JDK, so I suggest not to use it

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/StorageFileSystemProvider.java (hunk 2 above)
Comment (sbespalov) [reply strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/StorageFileSystemProvider.java]: done

**Comment by sbespalov** [comment]
> Would you mind rebasing? Thanks! :)

@carlspring done

**Comment by carlspring** [comment]
Thanks a lot, Sergey! :)

**Pull request merged**